Mathlib/Algebra/README.md

```diff
@@ -0,0 +1,17 @@
+# Algebra
+
+This folder contains algebra in a broad sense.
+
+## Algebraic hierarchy
+
+The basic algebraic hierarchy is split across a series of subfolders that each depend on the
+previous:
+* `Algebra.Notation` for basic algebraic notation
+* `Algebra.Group` for semigroups, monoids, groups
+* `Algebra.GroupWithZero` for monoids with a zero adjoined, groups with a zero adjoined
+* `Algebra.Ring` for semirings, rings
+* `Algebra.Field` for semifields, fields
+
+Files in earlier subfolders should not import files in later ones.
```

Comment:
Can this be enforced by a linter already, or are there still exceptions? Let's automate such tasks if we can :-)